cassiopeia/data/CassiopeiaTree.py

```diff
@@ -1144,6 +1168,37 @@ def leaves_in_subtree(self, node) -> List[str]:
 
         return self.__cache["subtree"][node]
 
+    def subset_tree_at_node(
```

Comment:
How do you feel about naming this function `clade` or simply `subtree`?

Reply:
I like it! I'm going to name it `subset_clade`, indicating clearly that we're subsetting the tree to obtain the clade at the node specified by the user.